flink-runtime/src/main/java/org/apache/flink/runtime/highavailability/nonha/embedded/EmbeddedJobResultStore.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.highavailability.nonha.embedded;
+
+import org.apache.flink.annotation.VisibleForTesting;
+import org.apache.flink.api.common.JobID;
+import org.apache.flink.runtime.highavailability.AbstractThreadsafeJobResultStore;
+import org.apache.flink.runtime.highavailability.JobResultStore;
+import org.apache.flink.runtime.highavailability.WithJobResult;
+import org.apache.flink.runtime.jobmaster.JobResult;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.concurrent.GuardedBy;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/** A thread-safe in-memory implementation of the {@link JobResultStore}. */
+public class EmbeddedJobResultStore extends AbstractThreadsafeJobResultStore {
+
+    private static final Logger LOG = LoggerFactory.getLogger(EmbeddedJobResultStore.class);
+
+    @GuardedBy("readWriteLock")
+    @VisibleForTesting
+    final Map<JobID, WithJobResult> dirtyJobResults = new HashMap<>();
+
+    @GuardedBy("readWriteLock")
+    @VisibleForTesting
+    final Map<JobID, WithJobResult> cleanJobResults = new HashMap<>();
+
+    @Override
+    public void createDirtyResultInternal(WithJobResult jobResultEntry) throws IOException {
```

Comment:
```suggestion
```
Additionally, the `IOExceptions` in the different method signatures can be removed. The `EmbeddedJobResultStore` does not throw any `IOException`. This would also cleanup the diff in the `EmbeddedJobResultStoreTest`.